examples/calculator/lib/main.dart

```diff
@@ -0,0 +1,261 @@
+  void setResult(CalcExpression resultExpression) {
+    _expressionStack.clear();
+    _expression = resultExpression;
+  }
+
+  void onNumberTap(int n) {
+    CalcExpression expression = _expression.appendDigit(n);
+    if (expression != null) {
+      setState(() {
+        pushExpression(expression);
+      });
+    }
+  }
+
+  void onPointTap() {
+    CalcExpression expression = _expression.appendPoint();
+    if (expression != null) {
+      setState(() {
+        pushExpression(expression);
+      });
+    }
+  }
+
+  void onPlusTap() {
+    CalcExpression expression = _expression.appendOperation(Operation.Addition);
+    if (expression != null) {
+      setState(() {
+        pushExpression(expression);
+      });
+    }
+  }
+
+  void onMinusTap() {
+    CalcExpression expression = _expression.appendMinus();
+    if (expression != null) {
+      setState(() {
+        pushExpression(expression);
+      });
+    }
+  }
+
+  void onMultTap() {
+    CalcExpression expression =
+        _expression.appendOperation(Operation.Multiplication);
+    if (expression != null) {
+      setState(() {
+        pushExpression(expression);
+      });
+    }
+  }
+
+  void onDivTap() {
+    CalcExpression expression = _expression.appendOperation(Operation.Division);
+    if (expression != null) {
+      setState(() {
+        pushExpression(expression);
+      });
+    }
+  }
+
+  void onEqualsTap() {
+    CalcExpression resultExpression = _expression.computeResult();
+    if (resultExpression != null) {
+      setState(() {
+        setResult(resultExpression);
+      });
+    }
+  }
+
+  void onDelTap() {
+    setState(() {
+      popCalcExpression();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+        appBar: new AppBar(title: new Text('Calculator')),
+        body: new Column(children: <Widget>[
+          // Give the key-pad 3/5 of the vertical space and the display
+          // 2/5.
+          new CalcDisplay(2, _expression.toString()),
+          new KeyPad(3, calcState: this)
+        ]));
+  }
+}
+
+class CalcDisplay extends StatelessWidget {
+  CalcDisplay(this._flex, this._contents);
+
+  int _flex;
+  String _contents;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Flexible(
+        flex: _flex,
+        child: new Center(child: new Text(_contents,
+            style: new TextStyle(color: Colors.black, fontSize: 24.0))));
+  }
+}
+
+class KeyPad extends StatelessWidget {
+  KeyPad(this._flex, {this.calcState});
+
+  final int _flex;
+  final _CalculatorState calcState;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Flexible(
+        flex: _flex,
+        child: new Row(children: <Widget>[
+          new MainKeyPad(calcState: calcState),
+          new OpKeyPad(calcState: calcState),
+        ]));
+  }
+}
+
+class MainKeyPad extends StatelessWidget {
+  final _CalculatorState calcState;
```

Comment:
I also didn't love the idea of passing the state object around like this.  But passing individual callbacks seems bad too here because I need a callback per button. So this solution seemed pragmatic.